core/src/main/java/org/apache/accumulo/core/clientImpl/TableMap.java

```diff
@@ -97,8 +96,8 @@ public TableMap(ClientContext context, ZooCache zooCache) {
         tableIdToNameBuilder.put(tableId, tableNameStr);
       }
     }
-    tableNameToIdMap = tableNameToIdBuilder.build();
-    tableIdToNameMap = tableIdToNameBuilder.build();
+    tableNameToIdMap = Collections.unmodifiableMap(tableNameToIdBuilder);
```

Comment:
People looking at this code after this change may not realize how important it is that these refs never escape.  Could add a comment or use an internal util method to make an immutable map, each of those would make the intention clear.